### PR TITLE
fix: integration tests for fileio runtimes.

### PR DESCRIPTION
**Pull request by RyanSkraba**
**Please check if the PR fulfills these requirements**

- [X] The commit message follows Talend standard
- [X] Tests for the changes have been added (for bug fixes / features)

**What is the current behavior?** (You can also link to an open issue here)

**What is the new behavior?**

Related to TCOMP-390, I added a module to `components-fileio` that should exactly duplicate the behaviour we expect from the REST API when accessing Dataset runtime methods.

This revealed some missing dependencies that didn't show up in unit tests in the `simplefileio-runtime` module, which were fixed.

**Does this PR introduce a breaking change?**

- [ ] Yes
- [X] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: components/components-fileio/simplefileio-runtime/pom.xml [hunk 1]
```diff
@@ -51,13 +51,25 @@
...
         <dependency>
             <groupId>org.apache.hadoop</groupId>
             <artifactId>hadoop-client</artifactId>
             <version>${hadoop.version}</version>
-            <scope>provided</scope>
+            <!--<scope>provided</scope>-->
+        </dependency>
+
+        <!-- To include in dependencies.txt. -->
+        <dependency>
```
Comment (sgandon) [inline components/components-fileio/simplefileio-runtime/pom.xml]: there was a guava issue that was fixed in https://github.com/Talend/components/pull/396/commits/3d4bb625977d4786d9aed8783813dfceb6832a9d
the hadoop version 2.7.0 does not use the broken api in guava 11.
So I think you do not have to force a version here.

**Comment by sgandon** [comment]
Also about the *provided* dependencies I think you can leave them as provided but then you need to add them in the test modules.

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/491/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/491/)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/527/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/527/)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/561/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/561/)

**Pull request merged**